amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/metrics/ShardConsumerMetricsReporter.java

```diff
@@ -75,4 +79,11 @@ public void setNumberOfDeaggregatedRecords(int numberOfDeaggregatedRecords) {
 		this.numberOfDeaggregatedRecords = numberOfDeaggregatedRecords;
 	}
 
+	public void unregister() {
+		((GenericMetricGroup) this.metricGroup).close();
```

Comment:
If we are relying on the `MetricGroup` being `GenericMetricGroup`, we should either:
- Require a `GenericMetricGroup` at the class/constructor
- Handle error case here where it is not a `GenericMetricGroup`

Reply:
I'll refactor both metric reporters to use a `GenericMetricGroup` and add a unit test for the error case.